### PR TITLE
Remove simple abci demo

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

* Ensure all the core info from docs on Simple ABCI demo is on Hello World demo
* Ensure all tests from Simple ABCI demo are on Hello World demo
* Remove all Simple ABCI related files and references
   * Service files
   * Agent files & tests
   * ABCI skill files (FSM app) & tests
   * Documentation
   * Misc. references (e.g., in Makefile)

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/agents/hello_world/aea-config.yaml [hunk 1]
```diff
@@ -21,8 +21,13 @@ protocols:
 - valory/http:1.0.0:bafybeigzqo2zaakcjtzzsm6dh4x73v72xg6ctk6muyp5uq5ueb7y34fbxy
 skills:
 - valory/abstract_abci:0.1.0:bafybeicog4eierjad4f542ubhe3ez7sxgrsna7t2e5pci2hncpq5vckw4e
+<<<<<<< HEAD
+- valory/abstract_round_abci:0.1.0:bafybeia5meor4p77p62khoddlqj4vg5rxiwi3djxc7feusgein6adi7t7m
+- valory/hello_world_abci:0.1.0:bafybeieq2w2x47o2qhtfvweqqvqaavkc37vtta6vmlaiqmoffifoqfkyre
+=======
 - valory/abstract_round_abci:0.1.0:bafybeihb5ihfhmbp3kjxp763f7675dltgrho6bj7yv7dkpp5aonczkjqom
 - valory/hello_world_abci:0.1.0:bafybeibz3i6hw2icoxfwex2l6b3buopsf55gw3pxvxcmyjf6b5bhkx6vry
+>>>>>>> refs/remotes/origin/main
```
Comment (Karrenbelt) [inline packages/valory/agents/hello_world/aea-config.yaml:30]: merge failure

File: packages/valory/agents/hello_world/tests/test_hello_world.py [hunk 2]
```diff
@@ -53,18 +55,113 @@
...
+
+# multiple agents terminating and restarting
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgentsTwoAgentRestarting(BaseHelloWorldABCITestCatchup):
+    """Test the hello_world_abci skill with four agents and two restarting."""
+
+    n_terminal = 2
+
+
+@pytest.mark.skip(reason="not working atm")
```
Comment (Karrenbelt) [inline packages/valory/agents/hello_world/tests/test_hello_world.py:137]: ??

Comment (jmoreira-valory) [reply packages/valory/agents/hello_world/tests/test_hello_world.py:137]: O_O

File: packages/valory/agents/hello_world/aea-config.yaml (hunk 1 above)
Comment (jmoreira-valory) [reply packages/valory/agents/hello_world/aea-config.yaml]: fixed

File: packages/valory/agents/hello_world/tests/test_hello_world.py (hunk 2 above)
Comment (DavidMinarsch) [reply packages/valory/agents/hello_world/tests/test_hello_world.py:137]: Why all these tests - was it like this for simple abci?

Comment (DavidMinarsch) [reply packages/valory/agents/hello_world/tests/test_hello_world.py:137]: Yes I see it was

**Review by DavidMinarsch — COMMENTED** [review]
What's the failing e2e test @Karrenbelt ?

File: packages/valory/agents/hello_world/tests/test_hello_world.py (hunk 2 above)
Comment (Karrenbelt) [reply packages/valory/agents/hello_world/tests/test_hello_world.py:137]: @DavidMinarsch why are these tests skipped? - 9511c4b93ca46a7a5f5a36459f2afdd1a18328c7

**Review by Karrenbelt — APPROVED** [review]
> What's the failing e2e test @Karrenbelt ?

```
2022-11-24T11:54:50.0018295Z Failed tests
2022-11-24T11:54:50.0018522Z Exit Code	Package
2022-11-24T11:54:50.0018786Z 1       	valory/agents/register_reset
```

when looking a little deeper, found these two. I am not quite sure why there is two of these and only a single failure.

1. 
    ```
    2022-11-24T11:25:24.0419370Z [2022-11-24 11:05:47,816][INFO] [agent_00000] HTTP Client has shutdown on port: 8000.
    2022-11-24T11:25:24.0419646Z Error: An error occurred during act of behaviour valory/register_reset_abci:0.1.0/RegisterResetAbciAppConsensusBehaviour:
    2022-11-24T11:25:24.0419760Z Traceback (most recent call last):
    2022-11-24T11:25:24.0419770Z 
    2022-11-24T11:25:24.0420200Z   File "/tmp/tmpli0ij7r5/agent_00000/vendor/valory/skills/abstract_round_abci/base.py", line 2439, in last_round_transition_timestamp
    2022-11-24T11:25:24.0425724Z     raise ValueError(
    2022-11-24T11:25:24.0425747Z 
    2022-11-24T11:25:24.0425976Z ValueError: Trying to access `last_round_transition_timestamp` while no transition has been completed yet.
    2022-11-24T11:25:24.0426008Z 
    2022-11-24T11:25:24.0426342Z [2022-11-24 11:05:47,926][INFO] [agent_00001] local-height = 69, remote-height=0
    2022-11-24T11:25:24.0426613Z [2022-11-24 11:05:47,927][INFO] [agent_00001] local height != remote height; retrying...
    2022-11-24T11:25:24.0426852Z [2022-11-24 11:05:47,943][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:24.0427426Z task: <Task pending name='Task-36' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpli0ij7r5/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:24.0427830Z [2022-11-24 11:05:47,943][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:24.0428405Z task: <Task pending name='Task-37' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpli0ij7r5/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:24.0428637Z [2022-11-24 11:05:47,943][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:24.0429198Z task: <Task pending name='Task-38' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpli0ij7r5/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:24.0429427Z [2022-11-24 11:05:47,943][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:24.0432036Z task: <Task pending name='Task-35' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpli0ij7r5/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    ```

2.
    ```
    2022-11-24T11:25:21.5282127Z [2022-11-24 10:55:27,740][INFO] [agent_00000] HTTP Client has shutdown on port: 8000.
    2022-11-24T11:25:21.5282489Z Error: An error occurred during act of behaviour valory/register_reset_abci:0.1.0/RegisterResetAbciAppConsensusBehaviour:
    2022-11-24T11:25:21.5282642Z Traceback (most recent call last):
    2022-11-24T11:25:21.5282653Z 
    2022-11-24T11:25:21.5290540Z   File "/tmp/tmpedwtwr4q/agent_00000/vendor/valory/skills/abstract_round_abci/base.py", line 2439, in last_round_transition_timestamp
    2022-11-24T11:25:21.5290816Z     raise ValueError(
    2022-11-24T11:25:21.5290842Z 
    2022-11-24T11:25:21.5291106Z ValueError: Trying to access `last_round_transition_timestamp` while no transition has been completed yet.
    2022-11-24T11:25:21.5291115Z 
    2022-11-24T11:25:21.5291508Z [2022-11-24 10:55:27,877][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:21.5292123Z task: <Task pending name='Task-36' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpedwtwr4q/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:21.5292416Z [2022-11-24 10:55:27,877][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:21.5293386Z task: <Task pending name='Task-37' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpedwtwr4q/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:21.5293676Z [2022-11-24 10:55:27,877][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:21.5294293Z task: <Task pending name='Task-38' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpedwtwr4q/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    2022-11-24T11:25:21.5294568Z [2022-11-24 10:55:27,877][ERROR] Task was destroyed but it is pending!
    2022-11-24T11:25:21.5295160Z task: <Task pending name='Task-35' coro=<TcpServerChannel.receive_messages() running at /tmp/tmpedwtwr4q/agent_00000/vendor/valory/connections/abci/connection.py:950> wait_for=<Future pending cb=[Task.task_wakeup()]>>
    ```
    
~~for which there is an issue open: #1430 ~~ EDIT: different issue, didn't realize.

File: packages/valory/agents/hello_world/tests/test_hello_world.py (hunk 2 above)
Comment (DavidMinarsch) [reply packages/valory/agents/hello_world/tests/test_hello_world.py:137]: As far as I can see they did not work fine and were ignored before

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**